src/main/scala/com/scylladb/migrator/validation/RowComparisonFailure.scala

```diff
@@ -64,6 +67,10 @@ object RowComparisonFailure {
             rightValue = rightMap.get(name)
 
             result = (rightValue, leftValue) match {
+              // All timestamp types need to be compared with a configured tolerance
+              case (Some(l: java.time.Instant), Some(r: java.time.Instant)) =>
+                (timestampMsTolerance <= 0 && l != r) ||
+                !(Math.abs(r.until(l, ChronoUnit.MILLIS)) <= timestampMsTolerance)
```

Comment:
I'm no expert in Scala, but would separating the pattern match be more readable? (Written off hand, without a compilation pass.)
```suggestion
              case (Some(l: java.time.Instant), Some(r: java.time.Instant)) if timestampMsTolerance <= 0 =>
                  l != r
               case (Some(l: java.time.Instant), Some(r: java.time.Instant)) =>                
                  Math.abs(r.until(l, ChronoUnit.MILLIS)) > timestampMsTolerance
```
Would be even nicer if Scala supported dependent typing `timestampMsTolerance` as a `NonNegativeInt/Double`. ^^

Reply:
I simplified it even more ;-)